src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -294,17 +303,239 @@ internal static ProcessInfo[] GetProcessInfos(Predicate<int> processIdFilter = n
+                        else if (processInfo.ProcessId == NtProcessManager.IdleProcessID)
+                        {
+                            processInfo.ProcessName = "Idle";
+                        }
+                        else
+                        {
+                            // for normal process without name, using the process ID. 
+                            processInfo.ProcessName = processInfo.ProcessId.ToString(CultureInfo.InvariantCulture);
+                        }
+                    }
+                    else
+                    {
+                        string processName = GetProcessShortName(Marshal.PtrToStringUni(pi.ImageName.Buffer, pi.ImageName.Length / sizeof(char)));
+                        processInfo.ProcessName = processName;
+                    }
+
+                    // get the threads for current process
+                    processInfos[processInfo.ProcessId] = processInfo;
+
+                    int threadInformationOffset = processInformationOffset + sizeof(SystemProcessInformation);
+                    for (int i = 0; i < pi.NumberOfThreads; i++)
+                    {
+                        ref readonly SystemThreadInformation ti = ref AsStruct<SystemThreadInformation>(data.Slice(threadInformationOffset));
+
+                        ThreadInfo threadInfo = new ThreadInfo();
+
+                        threadInfo._processId = (int)ti.ClientId.UniqueProcess;
+                        threadInfo._threadId = (ulong)ti.ClientId.UniqueThread;
+                        threadInfo._basePriority = ti.BasePriority;
+                        threadInfo._currentPriority = ti.Priority;
+                        threadInfo._startAddress = ti.StartAddress;
+                        threadInfo._threadState = (ThreadState)ti.ThreadState;
+                        threadInfo._threadWaitReason = NtProcessManager.GetThreadWaitReason((int)ti.WaitReason);
+
+                        processInfo._threadInfoList.Add(threadInfo);
+
+                        threadInformationOffset += sizeof(SystemThreadInformation);
+                    }
+                }
+
+                if (pi.NextEntryOffset == 0)
+                {
+                    break;
+                }
+                processInformationOffset += (int)pi.NextEntryOffset;
+            }
+
+            ProcessInfo[] temp = new ProcessInfo[processInfos.Values.Count];
+            processInfos.Values.CopyTo(temp, 0);
+            return temp;
+        }
+
+        // This function generates the short form of process name. 
+        //
+        // This is from GetProcessShortName in NT code base. 
+        // Check base\screg\winreg\perfdlls\process\perfsprc.c for details.
+        internal static string GetProcessShortName(String name)
+        {
+            if (String.IsNullOrEmpty(name))
+            {
+#if FEATURE_TRACESWITCH
+               Debug.WriteLineIf(Process._processTracing.TraceVerbose, "GetProcessInfos() - unexpected blank ProcessName");
+#endif
```

Comment:
Do we ever compile the library with this flag set?  Wondering if we can just delete this.